### PR TITLE
Feature: New Class MacAddress similar to IPAddress

### LINKED ISSUE #6658: MacAddress.h
### Related area

WiFi Library

### Hardware specification

ESP32, no specific hardware

### Is your feature request related to a problem?

I'm finding handling BSSID info to be a pain.  Parsing from string, printing to string, saving, checking if null, etc.  Is there a library for BSSIDs/MAC like IPAddress.cpp/h?

If not, I will create one patterned after IPAddress.cpp/h and submit a PR.

Just making sure I'm not duplicating work, and I'd like to know if this would be useful to others.

### Describe the solution you'd like

Say yes and I'll write a library and submit a PR.  I just don't want to waste my time and yours if a solution already exists.

### Describe alternatives you've considered

Searched Google extensively and didn't find functions to handle BSSID/mac within ESP32/arduino

### Additional context

_No response_

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
This adds new class `MacAddress` that allows for better handling of MAC and BSSID addresses. 
MacAddress also support  8-byte EUI-64 as noted in `esp_mac.h`.

Changes have been done on top of @mrengineer7777 PR #6667. Thanks David!

## Tests scenarios
Tested by running slightly modified test sketch by @mrengineer7777.
https://gist.github.com/P-R-O-C-H-Y/ab6b2443fc57bc12ec424f89493b9022

## Related links
Closes #6658

### CODE REVIEW

File: cores/esp32/MacAddress.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+String MacAddress::toString() const {
+    if(_type == MAC6) {
+        char buf[18];
+        sprintf(buf, "%02X:%02X:%02X:%02X:%02X:%02X",
+            _mac.bytes[0], _mac.bytes[1], _mac.bytes[2],
+            _mac.bytes[3], _mac.bytes[4], _mac.bytes[5]);
+        return String(buf);
+    } else {
+        char buf[24];
```
Comment (me-no-dev) [inline cores/esp32/MacAddress.cpp:135]: use one buf and one return, just change the sprintf in the if (and actually use `snprintf` to limit the size)

Comment (P-R-O-C-H-Y) [reply cores/esp32/MacAddress.cpp]: done

**Comment by me-no-dev** [comment]
@mrengineer7777 will you chime in to say if you are OK with this PR?

**Comment by lucasssvaz** [comment]
@P-R-O-C-H-Y Could you add a simple example sketch ?

**Pull request merged**